public/v4/apps/matrix-synapse.yml

```diff
@@ -0,0 +1,115 @@
+      dockerfileLines:
+        - FROM matrixdotorg/synapse:$$cap_synapse_version
+        - RUN mkdir /data/
+        - ENV SYNAPSE_SERVER_NAME=$$cap_appname.$$cap_root_domain
+        - ENV SYNAPSE_REPORT_STATS=$$cap_synapse_stats
+        - RUN /start.py generate
+        - RUN echo '\nserve_server_wellknown\072 true' >> /data/homeserver.yaml
+        - RUN echo "\nenable_registration\072 $$cap_enable_registration \nenable_registration_without_verification\072 true" >> /data/homeserver.yaml
+        - RUN echo "\nmax_upload_size\072 $$cap_max_upload_sizeM" >> /data/homeserver.yaml
+        - RUN sed -i '/^server_name.*/a public_baseurl\o072 "https://$$cap_appname.$$cap_root_domain"' /data/homeserver.yaml
+        - RUN sed -i 's/name\o072 sqlite3/name\o072 psycopg2/g' /data/homeserver.yaml
+        - RUN sed -i 's/database\o072 .*homeserver\.db/database\o072 synapse_db\n    user\o072 synapse_user\n    password\o072 $$cap_db_password\n    host\o072 srv-captain--$$cap_appname-db\n    port\o072 5432\n/g' /data/homeserver.yaml
+
+
+
+caproverOneClickApp:
+  variables:
+    - id: '$$cap_synapse_version'
+      label: Synapse Version
+      defaultValue: 'v1.63.1'
+      description: Check out their Docker page for the valid tags https://hub.docker.com/r/matrixdotorg/synapse/tags
+      validRegex: '/^v[0-9]{1,}|latest$/'
+    - id: '$$cap_time_zone'
+      label: Time zone
+      defaultValue: Europe/Berlin
+      description: The time zone of your server https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+      validRegex: '/.{1,}/'
+    - id: '$$cap_synapse_stats'
+      label: Report stats
+      defaultValue: 'no'
+      description: 'Report stats'
+      validRegex: '/^(yes|no)$/i'
+    - id: '$$cap_enable_registration'
+      label: Enable public registration
+      defaultValue: 'false'
+      description: >-
+        Set this to true if you want EVERYONE to be able to register.
+        It's recommended to set this to false and create users manually.
+      validRegex: '/^(true|false)$/i'
+    - id: '$$cap_max_upload_size'
+      label: Max upload size
+      defaultValue: '300'
+      description: >-
+        The maximum size of a file that can be uploaded, in megabytes.
+        
+        If you set this higher than 500, you will will also need to increase the nginx "client_max_body_size".
+      validRegex: '/^[0-9]{1,}$/'
+    - id: '$$cap_postgres_version'
+      label: Postgres version
+      defaultValue: '14'
+      description: >-
+        The version of postgres that you want to use.
+        You can find the valid versions here: https://hub.docker.com/r/postgres/postgres/tags
+      validRegex: '/^[0-9]{1,}|latest$/'
+    - id: '$$cap_db_password'
+      label: Database password
+      defaultValue: $$cap_gen_random_hex(20)
+      description: >-
+          The password for the Synapse database.
+      validRegex: '/^[0-9a-zA-Z]{1,}$/'
+  instructions:
+    start: >-
+      Matrix synapse server
```

Comment:
Please add more context here. A link to the website, a quick introduction etc... See [plex](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/plex.yml#L44-L47) for example